pilot/pkg/model/proxy_config.go

```diff
@@ -134,10 +135,12 @@ func mergeWithPrecedence(pcs ...*meshconfig.ProxyConfig) *meshconfig.ProxyConfig
 		// telemetry code does with shallowMerge?
 		proto.Merge(merged, pcs[i])
 		if pcs[i].GetConcurrency() != nil {
-			merged.Concurrency = pcs[i].GetConcurrency()
+			pcConcurrency := pcs[i].GetConcurrency()
+			merged.Concurrency = wrapperspb.Int32(pcConcurrency.Value)
 		}
 		if pcs[i].GetImage() != nil {
-			merged.Image = pcs[i].GetImage()
+			pcImage := pcs[i].GetImage()
+			merged.Image = &v1beta1.ProxyImage{ImageType: pcImage.ImageType}
```

Comment:
```suggestion
			merged.Image = proto.Clone(pcs[i].GetImage()).(*v1beta1.ProxyImage)
```

I am worried about us adding fields in the future an missing thing. We can do the same on L139 as well